msystems/services.py

```diff
@@ -1,26 +1,49 @@
 import logging
+from typing import List
+
+from core import datetime
+
 from django.db import transaction
+from django.db.models import Q
 from secrets import token_hex
 
 from core.models import User, InteractiveUser
 from core.services.userServices import create_or_update_user_districts
 from location.models import Location
+from policyholder.models import PolicyHolder, PolicyHolderUser
+
 logger = logging.getLogger(__name__)
 
 
 class SamlUserService:
-    
+    location = None
+
+    def __init__(self):
+        if not self.location:
+            self.location = Location.objects \
+                .prefetch_related('parent', 'parent__parent', 'parent__parent__parent') \
+                .get(code='MV01', validity_to__isnull=True)
+
     def login(self, username: str, user_data: dict):
         with transaction.atomic():
-            user = User.objects.prefetch_related(
-                'i_user').filter(username=username).first()
-            if not user:
-                user = self._create_user(username, user_data)
-            else:
-                self._update_user(user, user_data)
-            self._update_user_legal_entities(user, user_data)
-        
-            return user
+            try:
+                user = self._get_or_create_user(username, user_data)
+                self._update_user_legal_entities(user, user_data)
+                return user
+            except BaseException as e:
+                # Extra logging for the development, should be removed for any real data usage
+                # as it will put personal information in logs
+                logger.error("Successful SAML login handling failed, username=%s, user_data=%s", username,
```

Comment:
you can change this for logger.debug, then will be skipped in production environment

Reply:
I changed it to debug